doc/release/1.17.0-notes.rst

```diff
@@ -180,4 +180,11 @@ of NumPy functions on non-NumPy arrays, as described in `NEP 18`_. The feature
 was available for testing with NumPy 1.16 if appropriate environment variables
 are set, but is now always enabled.
 
+Do not lookup ``__buffer__`` attribute in `numpy.frombuffer`
+------------------------------------------------------------
+A python-level class could have defined the ``__buffer__`` attribute that
+would return the instance as a buffer. This was undocumented and flaky: if the
+``__buffer__`` attribute was a method it would not be called. This code was
+removed. If needed, use ``frombuffer(memoryview(obj), ...)`` instead.
+
```

Comment:
@charris is this appropriate? Should it be in a different section?

Reply:
I would put it in the `Compatibility notes` section and just say that it was non-functional and removed. We don't expect anyone to notice in practice, but if they do, there it is.

Reply:
adopted